extension/lib/logger.js

```diff
@@ -108,27 +104,43 @@ Logger.prototype._writeAsync = function Logger_writeAsync(aMessage, aCallback) {
   // For testing purposes send the message to stdout
   //dump(aMessage + '\n');
 
-  // Create an output stream to write to file
-  var foStream = Cc["@mozilla.org/network/file-output-stream;1"]
-                 .createInstance(Ci.nsIFileOutputStream);
-  foStream.init(this._file, 0x02 | 0x08 | 0x10, PERMS_FILE, foStream.DEFER_OPEN);
-
-  // Write asynchronously to buffer;
-  // Input and output streams are closed after write
-  var iStream = this._converter.convertToInputStream(aMessage);
-  NetUtil.asyncCopy(iStream, foStream, function (status) {
-    if (!components.isSuccessCode(status)) {
-      var errorMessage = new Error("Error while writing to file: " + status);
-      console.error(errorMessage);
+  function callback() {
+    if (typeof(aCallback) === "function") {
+      try {
+        aCallback();
+      } catch (e) {
+        Cu.reportError(e);
+      }
     }
+  }
 
-    if (typeof(aCallback) === "function") {
-      aCallback(status);
+  let array = this._encoder.encode(aMessage);
+  let promise = OS.File.open(this.file.path, {write: true})
+  .then(file => {
+    file.setPosition(0, OS.File.POS_END)
+    .then(file.write(array))
+    .then(
+      bytes => {
+        file.close();
+        callback();
+      },
+      error => {
+        file.close();
+        callback();
```

Comment:
As mentioned earlier please pass a status code to the callback, so calling code will know if saving the log entry was successful or not.